edb/lang/schema/functions.py

```diff
@@ -613,9 +613,32 @@ def _create_begin(self, schema, context):
         return_typemod = self.scls.get_return_typemod(schema)
         from_function = self.scls.get_from_function(schema)
         has_polymorphic = params.has_polymorphic(schema)
+        polymorphic_return_type = return_type.is_polymorphic(schema)
         named_only = params.find_named_only(schema)
 
-        if return_type.is_polymorphic(schema) and not has_polymorphic:
+        # Certain syntax is only allowed in "EdgeDB developer" mode,
+        # i.e. when populating std library, etc.
+        if not context.stdmode and not context.testmode:
+            if has_polymorphic or polymorphic_return_type:
+                raise ql_errors.EdgeQLError(
+                    f'cannot create `{signature}` function: '
+                    f'polymorphic types are not supported in '
```

Comment:
I'm thinking we went with slightly incorrect terminology with respect to `any` and friends.  IMO, `generics` or `generic types` is a more precise term.  Polymorphism is a property of functions and operators, so saying that something is a "polymorphic function" is OK.

Reply:
\cc @1st1  ^

Reply:
+1

Reply:
Wording updated.